Fails to compile without AOT (enable-minimal=profiler,debug,aot)

Mono fails to compile when AOT is disabled (enable-minimal=profiler,debug,aot).

## Steps to Reproduce

```
sh autogen.sh
./configure --enable-minimal=profiler,debug,aot
make
```

## Current Behavior

The function ```mono_aot_get_weak_field_indexes``` is used in mini-runtime.c and implemented in ```aot-runtime.c``` but not compiled in due to ```DISABLE_AOT``` being defined.

```
  CC       mono_boehm-main.o
  CCLD     mono-boehm
./.libs/libmini.a(libmini_la-mini-runtime.o): In function `mini_init':
/home/henry/Development/test/mono/mono/mini/mini-runtime.c:4041: undefined reference to `mono_aot_get_weak_field_indexes'
collect2: error: ld returned 1 exit status
make[4]: *** [mono-boehm] Error 1
make[4]: Leaving directory `/home/henry/Development/test/mono/mono/mini'
make[3]: *** [all] Error 2
make[3]: Leaving directory `/home/henry/Development/test/mono/mono/mini'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/henry/Development/test/mono/mono'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/henry/Development/test/mono'
make: *** [all] Error 2
```


## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**: 5.10.1.47 and also master from GitHub

--- a/mono/mini/aot-runtime.c
+++ b/mono/mini/aot-runtime.c
@@ -6179,4 +6179,10 @@ mono_aot_register_jit_icall (const char *name, gpointer addr)
 {
 }
 
+GHashTable *
+mono_aot_get_weak_field_indexes (MonoImage *image)
+{
+	return NULL;
+}
+
 #endif

